### ISSUE TITLE
rustdoc crashes on nightly nightly-2019-08-22

### ISSUE THREAD

**Issue by idubrov** [rust-lang/rust#63821]
`cargo doc` crashes on nightly (`nightly-2019-08-22`) and also on stable (`1.37.0`) with "nightly" features turned on via `RUSTC_BOOSTRAP=1`. It does not trigger on stable without `RUSTC_BOOTSTRAP=1` flag.

Here is the output I'm getting:
```
thread 'rustc' panicked at 'could not find markdown in source', src/libcore/option.rs:1166:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace.

error: internal compiler error: unexpected panic

note: the compiler unexpectedly panicked. this is a bug.

note: we would appreciate a bug report: https://github.com/rust-lang/rust/blob/master/CONTRIBUTING.md#bug-reports

note: rustc 1.39.0-nightly (e44fdf979 2019-08-21) running on x86_64-apple-darwin

error: Could not document `rustdoc-crash`.

Caused by:
  process didn't exit successfully: `rustdoc --edition=2018 --crate-name rustdoc_crash src/lib.rs -o /Users/idubrov/Projects/rustdoc-crash/target/doc --color always -L dependency=/Users/idubrov/Projects/rustdoc-crash/target/debug/deps --extern derive=/Users/idubrov/Projects/rustdoc-crash/target/debug/deps/libderive-aeb93296cea2c806.dylib` (exit code: 1)
```

I created a small repro case + README.md here: https://github.com/idubrov/rustdoc-crash (it only happens when there is a certain combination of procedural macros, so I cannot make it a single file).

**Comment by idubrov** [comment]
The source code is:

```rust
#[derive::first]
#[derive::second]
pub enum Boom {
    /// [Whatever]
    Bam,
}
```

with `derive` crate being a procedural macro crate with:
```rust
extern crate proc_macro;

#[proc_macro_attribute]
pub fn first(
  _attr: proc_macro::TokenStream,
  item: proc_macro::TokenStream,
) -> proc_macro::TokenStream {
  item
}

#[proc_macro_attribute]
pub fn second(
  _attr: proc_macro::TokenStream,
  item: proc_macro::TokenStream,
) -> proc_macro::TokenStream {
  // Note: if I return `item` as-is (same as `first` above), it resolves the issue!
  let mut out: proc_macro::TokenStream = proc_macro::TokenStream::new();
  out.extend(item);
  out
}
```

**Comment by idubrov** [comment]
Couple of things that are important here:

1. It only happens with two procedural macros.
2. Second macro has to do that `TokenStream::extend`.
3. Doc attribute has to be `[Whatever]`, does not reproduce with anything else.

**Comment by nox** [comment]
Just a heads-up: AFAIK if a bug only occurs on stable with `RUSTC_BOOTSTRAP=1` (I understand that it is not the case here), then it should probably be disregarded, because **`RUSTC_BOOTSTRAP=1` isn't supposed to be used when not bootstrapping the compiler**.

**Comment by nox** [comment]
The bug itself reminds me of https://github.com/rust-lang/rust/issues/46489.

**Comment by estebank** [comment]
@nox it happens on nightly as well. That detail was included to give context that evidently the ICE has been present for longer than 12 weeks, I believe.
